### PR TITLE
Setup GitHub Actions to run all tests

### PR DESCRIPTION
**Pull request by DeaglanBartlett**
Before this PR, only the `sbi` tests were being performed automatically upon pushes to the main branch. In this PR I have modified the workflow files so that all the tests can be performed (`sbi` and `pydelfi`) and the results combined to obtain a single summary report. The final output of this setup gives for the current version of the code

```
Wrote XML report to coverage.xml
Name                                Stmts   Miss  Cover   Missing
-----------------------------------------------------------------
ili/__init__.py                         8      0   100%
ili/dataloaders/__init__.py             1      0   100%
ili/dataloaders/loaders.py            100      5    95%   54, 63, 146, 258, 306
ili/embedding/__init__.py               1      0   100%
ili/embedding/fcn.py                   25      3    88%   10-11, 57
ili/inference/__init__.py               5      0   100%
ili/inference/pydelfi_wrappers.py      52      0   100%
ili/inference/runner_pydelfi.py        51      1    98%   92
ili/inference/runner_sbi.py           167     11    93%   102, 108, 129, 135, 182, 315-318, 336, 349
ili/utils/__init__.py                   2      0   100%
ili/utils/cosmology.py                 26      0   100%
ili/utils/dataset.py                   41     11    73%   31, 35, 54-56, 68-70, 83, 85, 87
ili/utils/import_utils.py               7      0   100%
ili/utils/samplers.py                  48      3    94%   125, 148-154
ili/validation/__init__.py              2      0   100%
ili/validation/metrics.py             229     58    75%   72, 83-86, 121-152, 266-290, 299, 316-317, 328, 350-351, 360, 379-380,  394-395, 450, 478-481, 495
ili/validation/runner.py               77     12    84%   77, 80, 90, 139-140, 144-151
-----------------------------------------------------------------
TOTAL                                 842    104    88%
```
so we can now see the total coverage for both backends. Obviously it would be nice to increase this to above 88%, but this is for a separate PR.

I have also made a couple of minor changes to the code to enable the `pydelfi` tests to pass.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -29,9 +29,77 @@ jobs:
           pip install -e .[sbi]
           git clone https://github.com/florpi/ili-summarizer.git
```
Comment (maho3) [inline .github/workflows/unit-tests.yml:30]: Remove this?

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -29,9 +29,77 @@ jobs:
...
+      - name: Setup MPI
+        uses: mpi4py/setup-mpi@v1
+        with:
+          mpi: 'openmpi'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          pip install -e .[pydelfi]
+          git clone https://github.com/florpi/ili-summarizer.git
```
Comment (maho3) [inline .github/workflows/unit-tests.yml:70]: Remove?

**Review by maho3 — CHANGES_REQUESTED** [review]
Looks good! Just need to remove ili-summarizer dependence since its no longer required. Then good for approve

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (DeaglanBartlett) [reply .github/workflows/unit-tests.yml]: Done

File: .github/workflows/unit-tests.yml (hunk 2 above)
Comment (DeaglanBartlett) [reply .github/workflows/unit-tests.yml]: Lines 67-69 are still needed, but I have removed the ili-summarizer installation.

**Review by maho3 — APPROVED** [review]

**Pull request merged**